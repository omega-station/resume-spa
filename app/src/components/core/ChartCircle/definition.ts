interface Core {}

export interface Props extends Core {
  /** Index */
  index: number;
  /** Label */
  label: string;
  /** Percentage */
  percent: number;
  /** Duration */
  duration?: number;
}

export interface PropsStyled extends Core {
  /** Animation values */
  animation: { [key: string]: number | string };
}
