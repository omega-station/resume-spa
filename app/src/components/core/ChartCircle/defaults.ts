import { Props } from './definition';

const defaults: Props = {
  index: 0,
  label: '',
  percent: 100,
  duration: 1500,
};

export default defaults;
