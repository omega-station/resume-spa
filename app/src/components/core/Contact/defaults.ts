import { Props } from './definition';

const defaults: Props = {
  hasDescription: false,
  hasTitle: true,
  isPostalWithCity: false,
};

export default defaults;
