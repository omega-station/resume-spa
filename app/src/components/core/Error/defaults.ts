import { Props } from './definition';

const defaults: Props = {
  icon: 'fas-exclamation-triangle',
  message: 'Something went wrong',
};

export default defaults;
