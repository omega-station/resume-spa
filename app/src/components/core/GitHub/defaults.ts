import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  ariaLabel: 'View source on GitHub',
  color: color.black,
  fill: color.white,
  mode: 'square',
  url: process.env.REACT_APP_URL_GITHUB,
};

export default defaults;
