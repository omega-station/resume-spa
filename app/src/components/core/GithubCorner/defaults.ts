import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  isSmall: false,
  ariaLabel: 'View source on GitHub',
  color: color.core.white,
  fill: color.core.black,
  url: 'https://github.com/omega-station/resume-spa',
};

export default defaults;
