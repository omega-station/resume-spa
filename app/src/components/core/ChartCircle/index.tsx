import _ from 'lodash';
import React from 'react';
// import { useCountUp } from 'react-countup';
import defaults from './defaults';
import { Props } from './definition';
import StyledChartCircle from './style';

const ChartCircle = (props: Props) => {
  const { index, label, percent }: Props = { ...defaults, ...props };
  const animation: { [key: string]: any } = {
    key: `${label.replace(/[ +]+/g, '-')}-${index}`,
    percent,
    delay: _.round(index * 0.275, 2),
    duration: 2,
    offset: 3,
  };

  // const { countUp } = useCountUp({
  //   delay: animation.delay,
  //   duration: animation.duration,
  //   end: percent,
  //   start: 0,
  //   useEasing: true,
  // });

  return (
    <StyledChartCircle animation={animation}>
      <svg viewBox="0 0 36 36" aria-hidden="true">
        <path d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831" />
        <path d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831" />
        {/* <text>{countUp}%</text> */}
        <text x="50%" y="60%">
          {percent}%
        </text>
      </svg>
      <span>{label}</span>
    </StyledChartCircle>
  );
};

export default ChartCircle;
