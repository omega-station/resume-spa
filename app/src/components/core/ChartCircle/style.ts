import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { color } from '../../../utility/constant';
import { PropsStyled as Props } from './definition';

const getColor = (percent: any): string => {
  switch (true) {
    case percent >= 85:
      return color.red.medium;
    case percent >= 75:
      return color.red.medium2;
    default:
      return color.red.dark;
  }
};

const StyledChartCircle = styled.div<Props>`
  ${props => css`
    flex-direction: column;
    justify-content: center;
    align-items: center;

    svg {
      display: block;
      margin: 0 auto;
      max-width: 85%;
      max-height: 250px;

      path {
        fill: none;

        &:first-of-type {
          stroke: ${color.grey.light2};
          stroke-width: 4;
        }

        &:last-of-type {
          stroke: ${getColor(props.animation.percent)};
          stroke-dasharray: 0 100;
          stroke-width: 3;
          animation: ${`anim-chart-${props.animation.key} ${props.animation.duration * 1000}ms ${props.animation.delay * 1000}ms ease-out forwards`};

          @keyframes anim-chart-${props.animation.key} {
            from {
              stroke-linecap: round;
            }
            to {
              stroke-dasharray: ${props.animation.percent - props.animation.offset} 100;
              stroke-linecap: round;
            }
          }
        }
      }

      text {
        fill: ${color.grey.dark};
        font-size: 0;
        text-anchor: middle;
        animation: ${`anim-text-${props.animation.key} ${props.animation.duration * 500}ms ${props.animation.delay * 1000}ms forwards`};

        @keyframes anim-text-${props.animation.key} {
          to {
            font-size: 0.65rem;
            font-size: 0.75rem;
          }
        }
      }
    }

    span {
      overflow: hidden;
      display: block;
      width: 100%;
      height: 40px;
      margin: 12px 0;
      color: ${color.black};
      font-size: 1.35rem !important;
      font-weight: 500;
      line-height: 1em;
      text-align: center;
    }
  `};
`;

export default StyledChartCircle;
