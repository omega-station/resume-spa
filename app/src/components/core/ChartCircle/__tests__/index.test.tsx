import { MockedProvider } from '@apollo/react-testing';
import { act, cleanup, render } from '@testing-library/react';
import React from 'react';
import ChartCircle from '..';
import defaults from '../defaults';

afterEach(cleanup);

describe('ChartCircle', () => {
  it('should render with label={default} and percent={default} props', () => {
    const component = render(<ChartCircle label={defaults.label} percent={defaults.percent} />);
    expect(component).toMatchSnapshot();
  });

  it('should render with label={default}, percent={default}, and color={default} props', () => {
    const component = render(<ChartCircle label={defaults.label} percent={defaults.percent} color={defaults.color} />);
    expect(component).toMatchSnapshot();
  });
});
