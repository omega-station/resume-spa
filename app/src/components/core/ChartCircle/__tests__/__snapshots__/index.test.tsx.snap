// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartCircle should render with index={default}, label={default} and percent={default} props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
  -webkit-animation: anim-chart--0 2000ms 0ms ease-out forwards;
  animation: anim-chart--0 2000ms 0ms ease-out forwards;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0;
  text-anchor: middle;
  -webkit-animation: anim-text--0 1000ms 0ms forwards;
  animation: anim-text--0 1000ms 0ms forwards;
}

.emotion-0 span {
  overflow: hidden;
  display: block;
  width: 100%;
  height: 40px;
  margin: 12px 0;
  color: rgb(0,0,0);
  font-size: 1.35rem !important;
  font-weight: 500;
  line-height: 1em;
  text-align: center;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <svg
          aria-hidden="true"
          viewBox="0 0 36 36"
        >
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <text
            x="50%"
            y="60%"
          >
            100
            %
          </text>
        </svg>
        <span />
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
  -webkit-animation: anim-chart--0 2000ms 0ms ease-out forwards;
  animation: anim-chart--0 2000ms 0ms ease-out forwards;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0;
  text-anchor: middle;
  -webkit-animation: anim-text--0 1000ms 0ms forwards;
  animation: anim-text--0 1000ms 0ms forwards;
}

.emotion-0 span {
  overflow: hidden;
  display: block;
  width: 100%;
  height: 40px;
  margin: 12px 0;
  color: rgb(0,0,0);
  font-size: 1.35rem !important;
  font-weight: 500;
  line-height: 1em;
  text-align: center;
}

<div>
    <div
      class="emotion-0"
    >
      <svg
        aria-hidden="true"
        viewBox="0 0 36 36"
      >
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <text
          x="50%"
          y="60%"
        >
          100
          %
        </text>
      </svg>
      <span />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChartCircle should render with label={default} and percent={default} props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0.65rem;
  text-anchor: middle;
  -webkit-transform: translate(18px,21px);
  -ms-transform: translate(18px,21px);
  transform: translate(18px,21px);
}

.emotion-0 span {
  color: rgb(0,0,0);
  font-size: 1.35rem;
  font-weight: 500;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <svg
          aria-hidden="true"
          viewBox="0 0 36 36"
        >
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
        </svg>
        <span />
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0.65rem;
  text-anchor: middle;
  -webkit-transform: translate(18px,21px);
  -ms-transform: translate(18px,21px);
  transform: translate(18px,21px);
}

.emotion-0 span {
  color: rgb(0,0,0);
  font-size: 1.35rem;
  font-weight: 500;
}

<div>
    <div
      class="emotion-0"
    >
      <svg
        aria-hidden="true"
        viewBox="0 0 36 36"
      >
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
      </svg>
      <span />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ChartCircle should render with label={default}, percent={default}, and color={default} props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0.65rem;
  text-anchor: middle;
  -webkit-transform: translate(18px,21px);
  -ms-transform: translate(18px,21px);
  transform: translate(18px,21px);
}

.emotion-0 span {
  color: rgb(0,0,0);
  font-size: 1.35rem;
  font-weight: 500;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <svg
          aria-hidden="true"
          viewBox="0 0 36 36"
        >
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
          <path
            d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
          />
        </svg>
        <span />
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  display: block;
  margin: 0 auto;
  max-width: 85%;
  max-height: 250px;
}

.emotion-0 svg path {
  fill: none;
}

.emotion-0 svg path:first-of-type {
  stroke: rgb(200,200,200);
  stroke-width: 4;
}

.emotion-0 svg path:last-of-type {
  stroke: rgb(173,42,36);
  stroke-dasharray: 0 100;
  stroke-width: 3;
}

.emotion-0 svg text {
  fill: rgb(60,60,60);
  font-size: 0.65rem;
  text-anchor: middle;
  -webkit-transform: translate(18px,21px);
  -ms-transform: translate(18px,21px);
  transform: translate(18px,21px);
}

.emotion-0 span {
  color: rgb(0,0,0);
  font-size: 1.35rem;
  font-weight: 500;
}

<div>
    <div
      class="emotion-0"
    >
      <svg
        aria-hidden="true"
        viewBox="0 0 36 36"
      >
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <path
          d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
        />
      </svg>
      <span />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
